proc/proc_darwin.go

```diff
@@ -316,6 +362,18 @@ func (dbp *Process) trapWait(pid int) (*Thread, error) {
 		case 0:
 			return nil, fmt.Errorf("error while waiting for task")
 		}
+		
+		// In macOS 10.12.1 if we received a notification for a task other than
+		// the inferior's task and the inferior's task is no longer valid, this
+		// means inferior called execve and its task_t changed.
+		if dbp.os.task != task && C.task_is_valid(dbp.os.task) == 0 {
+			fmt.Printf("task %d -> %d\n", dbp.os.task, task)
```

Comment:
nit: remove this print lines, or use proper logging which we can enable / disable via log flag.